src/model/generalActions.js

```diff
@@ -190,36 +220,57 @@ const reactToGeneralAction = (model) =>
         break;
 
       case 'LEAVE_CANVAS':
-        if ((model.mode === 'FREEDRAW' || model.mode === 'POLYDRAW' || model.mode === 'RECTANGLE') 
-            && model.activityInProgress === true){
-          model.activityInProgress = false;
-          model.clicks = 0;
+        if (model.activityInProgress === false) {
+          break;
+        }
+
+        model.activityInProgress = false;
+        model.clicks = 0;
+
+        if (model.mode === 'FREEDRAW' || model.mode === 'POLYDRAW' || model.mode === 'RECTANGLE'){
           model.annotations.pop();
         }
+
+        if ((model.mode === 'SELECT') && model.activityInProgress === true) {
```

Comment:
how would activityinprogress not be true given code breaks out above if its false